src/main/java/com/o19s/es/ltr/logging/LoggingFetchSubPhase.java

```diff
@@ -32,35 +32,35 @@
 import org.elasticsearch.common.document.DocumentField;
 import org.elasticsearch.search.SearchHit;
 import org.elasticsearch.search.fetch.FetchSubPhase;
-import org.elasticsearch.search.fetch.subphase.InnerHitsContext;
-import org.elasticsearch.search.internal.SearchContext;
+import org.elasticsearch.search.fetch.FetchSubPhaseProcessor;
+import org.elasticsearch.search.fetch.FetchContext;
 import org.elasticsearch.search.rescore.QueryRescorer;
 import org.elasticsearch.search.rescore.RescoreContext;
 
 import java.io.IOException;
 import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.Collections;
-import java.util.Comparator;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 
 public class LoggingFetchSubPhase implements FetchSubPhase {
     @Override
-    public void hitsExecute(SearchContext context, SearchHit[] hits) throws IOException {
+    public FetchSubPhaseProcessor getProcessor(FetchContext context) throws IOException {
         LoggingSearchExtBuilder ext = (LoggingSearchExtBuilder) context.getSearchExt(LoggingSearchExtBuilder.NAME);
         if (ext == null) {
-            return;
+            return null;
         }
 
-        // Use a boolean query with all the models to log
-        // This way we reuse existing code to advance through multiple scorers/iterators
         BooleanQuery.Builder builder = new BooleanQuery.Builder();
         List<HitLogConsumer> loggers = new ArrayList<>();
         Map<String, Query> namedQueries = context.parsedQuery().namedFilters();
-        if (!(context instanceof InnerHitsContext.InnerHitSubContext)) {
+
+
+        //TODO: Need help verifying the type of the context, SearchContext was moved inside of FetchContext
+        // context instanceof InnerHitsContext.InnerHitSubContext seems to no longer be accessible
+        if (namedQueries.size() > 0) {
```

Comment:
@nomoa I can't check the instance of the SearchContext anymore as it's wrapped privately in a FetchContext.  Any ideas?  The check on namedQueries size makes the tests happy but I don't know if that catches all cases.